libraries/AP_Mission/AP_Mission_Commands.cpp

```diff
@@ -0,0 +1,112 @@
+        sre->do_set_relay(cmd.content.relay.num, cmd.content.relay.state);
+        break;
+
+    case MAV_CMD_DO_REPEAT_SERVO:
+        sre->do_repeat_servo(cmd.content.repeat_servo.channel,
+                             cmd.content.repeat_servo.pwm,
+                             cmd.content.repeat_servo.repeat_count,
+                             cmd.content.repeat_servo.cycle_time * 1000.0f);
+        break;
+
+    case MAV_CMD_DO_REPEAT_RELAY:
+        sre->do_repeat_relay(cmd.content.repeat_relay.num,
+                             cmd.content.repeat_relay.repeat_count,
+                             cmd.content.repeat_relay.cycle_time * 1000.0f);
+        break;
+    default:
+        break;
+    }
+
+    return true;
+}
+
+bool AP_Mission::start_command_camera(const AP_Mission::Mission_Command& cmd)
+{
+    AP_Camera *camera = AP::camera();
+    if (camera == nullptr) {
+        return true;
+    }
+
+    switch (cmd.id) {
+    case MAV_CMD_DO_CONTROL_VIDEO:                      // Control on-board camera capturing. |Camera ID (-1 for all)| Transmission: 0: disabled, 1: enabled compressed, 2: enabled raw| Transmission mode: 0: video stream, >0: single images every n seconds (decimal)| Recording: 0: disabled, 1: enabled compressed, 2: enabled raw| Empty| Empty| Empty|
+        break;
+
+    case MAV_CMD_DO_DIGICAM_CONFIGURE:                  // Mission command to configure an on-board camera controller system. |Modes: P, TV, AV, M, Etc| Shutter speed: Divisor number for one second| Aperture: F stop number| ISO number e.g. 80, 100, 200, Etc| Exposure type enumerator| Command Identity| Main engine cut-off time before camera trigger in seconds/10 (0 means no cut-off)|
+        camera->configure(
+            cmd.content.digicam_configure.shooting_mode,
+            cmd.content.digicam_configure.shutter_speed,
+            cmd.content.digicam_configure.aperture,
+            cmd.content.digicam_configure.ISO,
+            cmd.content.digicam_configure.exposure_type,
+            cmd.content.digicam_configure.cmd_id,
+            cmd.content.digicam_configure.engine_cutoff_time);
+        break;
+
+    case MAV_CMD_DO_DIGICAM_CONTROL:                    // Mission command to control an on-board camera controller system. |Session control e.g. show/hide lens| Zoom's absolute position| Zooming step value to offset zoom from the current position| Focus Locking, Unlocking or Re-locking| Shooting Command| Command Identity| Empty|
+        camera->control(
+            cmd.content.digicam_control.session,
+            cmd.content.digicam_control.zoom_pos,
+            cmd.content.digicam_control.zoom_step,
+            cmd.content.digicam_control.focus_lock,
+            cmd.content.digicam_control.shooting_cmd,
+            cmd.content.digicam_control.cmd_id);
+        break;
+
+    case MAV_CMD_DO_SET_CAM_TRIGG_DIST:
+        camera->set_trigger_distance(cmd.content.cam_trigg_dist.meters);
+        break;
+
+    default:
+        return true;
```

Comment:
Nitpick: return false, because we didn't handle the command so we should start the next one immediately? (I realize that do commands we skim through, but if we get in here and don't handle the command it may have been a nav command by accident... (This comment applies in a couple of places, I just need to think it through a bit further).